lib/flappy_translator.dart

```diff
@@ -48,31 +51,52 @@ const List<String> RESERVED_WORDS = [
 ];
 
 class FlappyTranslator {
-  void generate(String filePath, {String targetPath = ""}) async {
-    final File file = File(filePath);
-
+  void generate(
+    String inputFilePath, {
+    String outputDir,
+    String fileName,
+    String className,
+    String delimiter,
+    int startIndex,
+    bool dependOnContext,
+    bool useSingleQuotes,
+  }) async {
+    final File file = File(inputFilePath);
     if (!file.existsSync()) {
-      FlappyLogger.logError("File $filePath does not exist");
+      FlappyLogger.logError("File $inputFilePath does not exist");
       return;
     }
 
-    String template = templateString;
+    // if null has been passed in, ensure that vars are given default values
+    outputDir ??= defaultOutputDirectory;
+    fileName ??= defaultFileName;
+    className ??= defaultClassName;
+    delimiter ??= defaultDelimiter;
+    startIndex ??= defaultStartIndex;
+    dependOnContext ??= defaultDependOnContext;
+    useSingleQuotes ??= defaultUseSingleQuotes;
```

Comment:
Why do you not use the default parameters on the constructor to initialize fields with this default values ?

Reply:
In `bin/flappy_translator` the settings from `pubspec.yaml` may be null if the user did not specify them. Thus in the method call `flappyTranslator.generate()`, null could be passed in a value. In this case, even if the method defines a default value, null be be chosen as this is the value passed in. Hence the decision to re-initialize the variables at the begining of the method. Maybe there is a better way to achieve this, but defining the default values twice in `bin/flappy_translator` and `lib/flappy_translator` seems overkill, for instance:

```
flappyTranslator.generate(
    ...
    fileName: settings[YamlArguments.fileName] ?? defaultFileName,
    ...

class FlappyTranslator {
  void generate(
    String inputFilePath, {
    ...
    String fileName = defaultFileName,
    ...
```